transaction/core/src/blockchain/block_version.rs

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2018-2021 The MobileCoin Foundation
+
+use core::{fmt, hash::Hash, ops::Deref};
+use mc_crypto_digestible::Digestible;
+use serde::{Deserialize, Serialize};
+
+/// A block version number
+#[derive(
+    Clone, Copy, Debug, Eq, PartialEq, PartialOrd, Ord, Serialize, Deserialize, Digestible, Hash,
+)]
+pub struct BlockVersion(pub u32);
```

Comment:
This should be private.

Reply:
The thing is, if it is private then I can't use the nice syntax `BlockVersion(1)` to construct it

Reply:
Right, I'm saying you can have BlockVersion > MAX if you do that, which probably isn't valid.